lib/aws_ssm_provider.ex

```diff
@@ -32,64 +37,71 @@ defmodule AwsSsmProvider do
     end
   end
 
-  defp set_vars([], _), do: :ok
-
-  defp set_vars([head | tail], app_name) do
-    val = translate_values(head["Value"])
-
-    head["Name"]
-    |> String.split("/", trim: true)
-    |> Enum.map(&String.to_atom/1)
-    |> remove_prefix_keys
-    |> set_app_name(app_name)
-    |> persist(val)
-
-    set_vars(tail, app_name)
+  defp parse_ssm_config(ssm_configs, app_name) do
+    configs =
+      ssm_configs
+      |> Enum.map(&lex_ssm_entry(&1, app_name))
+      |> Enum.map(&build_config_branch/1)
+      |> Enum.reduce([], fn new_config, prev_configs ->
+        Config.Reader.merge(prev_configs, new_config)
+      end)
+
+    {:ok, configs}
+  end
+
+  @type lexed_entry :: {path :: [atom()], value :: term()}
```

Comment:
Comment: I like having the `@type` defined right about the typespec. Much easier than jumping around the file 🏅